portal-api/src/scheduler/jobs/send-report-due-emails-job/index.test.js

```diff
@@ -0,0 +1,146 @@
+const { subDays } = require('date-fns');
+const { produce } = require('immer');
+const sendReportDueEmailsJob = require('.');
+const api = require('../../../v1/api');
+const externalApi = require('../../../external-api/bank-holidays');
+const sendEmail = require('../../../external-api/send-email');
+const MOCK_BANKS = require('../../../../test-helpers/mock-banks');
+const MOCK_UTILISATION_REPORT = require('../../../../test-helpers/mock-utilisation-reports');
+const { EMAIL_TEMPLATE_IDS } = require('../../../constants');
+
+jest.mock('../../../v1/api');
+jest.mock('../../../external-api/bank-holidays');
+jest.mock('../../../external-api/send-email', () => jest.fn());
+
+console.error = jest.fn();
+console.warn = jest.fn();
+console.info = jest.fn();
+
+const sendReportDueEmailsJobTask = sendReportDueEmailsJob.init().task;
+
+const NOV_2023_REPORT_DUE_DATE = new Date('2023-11-15');
+
+describe('sendReportDueEmailsJob', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    jest.useRealTimers();
+  });
+
+  it('does not send emails when report is not due today', async () => {
+    // Arrange
+    const today = subDays(NOV_2023_REPORT_DUE_DATE, 1);
+    jest.useFakeTimers().setSystemTime(today);
+
+    externalApi.getBankHolidayDatesForRegion.mockResolvedValue([]);
+
+    // Act
+    await sendReportDueEmailsJobTask();
+
+    // Assert
+    expect(sendEmail).not.toHaveBeenCalled();
+    expect(console.info).toHaveBeenCalledWith(expect.stringContaining('report is not due today'));
+  });
+
+  it('does not send an email when the bank has already submitted their report', async () => {
+    // Arrange
+    jest.useFakeTimers().setSystemTime(NOV_2023_REPORT_DUE_DATE);
+
+    externalApi.getBankHolidayDatesForRegion.mockResolvedValue([]);
+
+    api.getAllBanks.mockResolvedValue([MOCK_BANKS.HSBC]);
+
+    const existingReport = {
+      ...MOCK_UTILISATION_REPORT,
+      // 'month' should be 1-indexed, Date.getMonth() is 0-indexed, so this sets 'month' to the previous month
+      month: NOV_2023_REPORT_DUE_DATE.getMonth(),
+      year: NOV_2023_REPORT_DUE_DATE.getFullYear(),
```

Comment:
I've made some comments about `MOCK_UTILISATION_REPORT` below, but I think those changes can be implemented to improve how `existingReport` is defined